### ISSUE TITLE
Should rootfs-packages exist?

### ISSUE THREAD

**Issue by woodenshoe-wi** [puppylinux-woof-CE/woof-CE#1176]
To keep from getting too off topic I am starting a new issue.

It is my understanding that the reason for rootfs-packages is to hold scripts that are optional, if they are mandatory they would be in rootfs-skeleton instead.

In kernel-kit various bits and pieces are pulled in from git repos.

> rootfs-packages must at least consist of packages that are not recorded...

If they are not supposed to be recorded then I guess they better remain part of woof-CE, but if woof-CE supported including packages from outside git repos there would be no need to get anyone to process or upload anything.  As soon as they updated their git repo the changes would take effect.

> And this is how the problem started... allowing non-woofce packages
when a woofce replacement was already there.

If the idea is not to allow more independence but to control the packages more closely I don't want to waste my time trying to implement something that is not wanted.

Regarding the "generic names", if the package list format would be changed in the future I would also request that an "arch" field would be added so that 32bit and 64bit packages could coexist in the same package list...

**Comment by mavrothal** [comment]
Not sure where the quotes are coming from but the original rootfs-packages goal was to get all the "common" pets in the git and leave ibiblio for compiled packages only. 
So the idea was to bring all those "puppy scripts" and "dialog apps" *with their developers* into woof so would be easier to develop and update without the need of ibiblio.
Of course this did not go very far, so I'm not sure if rootfs-packages should exist or how/to what should it be transformed at this point.

**Comment by woodenshoe-wi** [comment]
Sorry, I forgot to mention that this conversation started on issue #1167

**Comment by wdlkmpx** [comment]
Pet packages can not be updated - fact. 

The woofce apps keep getting fixes until they become usable.. fact.

Stuff breaks with pet pkgs not properly integrated i.e: old jwmconfigs.

It's not that hard to explain why centralization is needed.

Consistency in the end may be something silly, i can certainly do everything on my own and when i want to implement or make something work, i don't need anyone to help me.

**Comment by wdlkmpx** [comment]
Incompability with other/older woof build systems

In the process of improving code, applying a vision or changing stuff, incompatibilities happen, i don't know why, but i get impression that what's on ibiblio is basically compatible stuff with other woofs, or compat-distro version specific... pet pkgs fixed in certain threads for a specific pup.. till the point the it becomes a struggle between pups.

This incoherent behavior was the norm, it may be the way go.

**Comment by woodenshoe-wi** [comment]
Then there are comments like this one from mavrothal
https://github.com/puppylinux-woof-CE/woof-CE/pull/1168#issuecomment-381432794

> I think would be better to have better packages (pets) rather than trying to fix everything in woof.

The part of the pet that I wanted moved into woof-CE consists of all the MIME-types run actions that go into /etc/xdg/rox.sourceforge.net/, the /root/Choices stuff and the pinstall.sh

I guess the pinstall.sh is harmless, but do you want to repackage the pet just to change it?

I have found that I enjoy working on the ROX source code much more than trying to merge changes into woof-CE.  So the best solution for me is probably to just work on what I like.

**Comment by wdlkmpx** [comment]
Once upon a time the jwmconfig was part of the jwm pkg.

But then it was moved to a separate pkg. When it comes to icons and
stuff, mimetypes and general config.. the rox_config should born.

But this will conflict with current pet pkgs, as a rootfs-package it
will get overwritten. It's really up the contributor, i forced pcmanfm
support into the system, just so i can avoid rox as the file manager,
avoid the torture and keep contributing.

The 1st step is create the rox_config rootfs-package, 2nd: make the
impossible, try to deal with other unresponsive devs.. in the git repo
and in the pet repos.

Hell is on earth..

**Comment by woodenshoe-wi** [comment]
That is where a pet template has an advantage, it would overwrite whatever is in the pet...  but that would be trying to "fix everything in woof"...

**Comment by wdlkmpx** [comment]
For a future rox_config ignoring what everyone else has to say
(nothing basically) and at same time forcing a more up-to-date and
patched rox (although no amount of patches can fix it enough
https://github.com/jun7/rox-filer/issues/137 )... this is how it could
be done...

1) woodenshoe-wi adss rox_config
2) woodenshoe-wi patches rox to only depend on gtk2 and x... i see
librsvg in https://www.archlinux.org/packages/community/x86_64/rox/
... there is nanorsvg and other stuff technosaurus knows about, that
can be adapted and linked statically to rox...
3) rox is ready to be a common app
4) woodenshoe-wi asks peebee to upload the patched and generic rox
(i686/x86_64/arm)
5) rox_config is always enabled or else..

**Comment by woodenshoe-wi** [comment]
No,
1. woodenshoe-wi does not "adss" rox_config, If it is left to me it is easier to just make two pets, one with the extra stuff from the existing template and one "stand-alone" version safe to use in an existing install.
2. woodenshoe-wi is working on a pupified woodenshoe-wi fork that is a modified combination of dtomas's fork (copy to clipboard = ctrl-c, cut to clipboard = ctrl-x) seirios's fork (xattr browser), some cherry picked commits from jun7, the patches from the petbuild and my own fixes.
3. I already was able to compile the old version in the petbuild as a common app, hopefully the new version will too.
4. woodenshoe-wi asked peebee NOT to upload the new version until it gets proper testing.
5. See number 1.

**Comment by wdlkmpx** [comment]
Those were just suggestions based on my experience on how to do
things. Of course i don't expect everyone to do the same. I'm myself
not adding any pet pkgs outside my domain or pkgs templates for pet
pkgs (my invention), i also removed the ones i added, of course
somebody can revert the commit if they see something is broken because
of that, but that's my way of saying things have changed.
